File: javascript/pascal.js
var generate = function(numRows) {
  const pascalsTriangle = [];
 for (let i = 0; i < numRows; i++) {
   let presentArray = [];
   for (let j = 0; j <= i; j++) {
     if (j == 0) {
       presentArray.push(1);
     } else {
       if (pascalsTriangle[i - 1]) {
         let num1;
         let num2;
         pascalsTriangle[i - 1][j]
           ? (num1 = pascalsTriangle[i - 1][j])
           : (num1 = 0);
         pascalsTriangle[i - 1][j - 1]
           ? (num2 = pascalsTriangle[i - 1][j - 1])
           : (num2 = 0);
         presentArray.push(num1 + num2);
       } else {
         presentArray.push(0);
       }
     }
   }
   pascalsTriangle.push(presentArray);
 }
 return pascalsTriangle;
};

File: python/pascal.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        a=[]

        for i in range(0,numRows):
            a.append([0 for x in range(i+1)])
        for i in range(0,numRows):
            for j in range(0,i+1):
                if(i==0):
                    a[i][j]=1
                elif(i==1):
                    a[i][j]=1
                elif( i>1 and ((j==i) or (j==0))):
                     a[i][j]=1
                else:
                     a[i][j]=a[i-1][j-1]+a[i-1][j]

        return (a)
